website/orders/tests/test_models.py

```diff
@@ -0,0 +1,558 @@
+        models.Order.objects.create(user=self.normal_user, shift=self.shift, product=self.product_2)
+        self.assertTrue(product.user_can_order_amount(self.normal_user, self.shift))
+
+    def test_product_user_order_amount_no_restriction(self):
+        product = models.Product.objects.create(
+            name="Test no restriction", max_allowed_per_shift=None, current_price=0.8
+        )
+        self.assertTrue(product.user_can_order_amount(self.normal_user, self.shift))
+        self.assertTrue(product.user_can_order_amount(self.normal_user, self.shift, amount=2))
+        self.assertTrue(product.user_can_order_amount(self.normal_user, self.shift, amount=3))
+        models.Order.objects.create(user=self.normal_user, shift=self.shift, product=product)
+        self.assertTrue(product.user_can_order_amount(self.normal_user, self.shift))
+        self.assertTrue(product.user_can_order_amount(self.normal_user, self.shift, amount=2))
+        models.Order.objects.create(user=self.normal_user, shift=self.shift, product=self.product_2)
+        self.assertTrue(product.user_can_order_amount(self.normal_user, self.shift, amount=2))
+
+    def test_product_user_max_order_amount(self):
+        product = models.Product.objects.create(name="Test no restriction", max_allowed_per_shift=2, current_price=0.8)
+        self.assertEqual(product.user_max_order_amount(self.normal_user, self.shift), 2)
+        models.Order.objects.create(user=self.normal_user, shift=self.shift, product=product)
+        self.assertEqual(product.user_max_order_amount(self.normal_user, self.shift), 1)
+        models.Order.objects.create(user=self.normal_user, shift=self.shift, product=self.product_2)
+        self.assertEqual(product.user_max_order_amount(self.normal_user, self.shift), 1)
+
+    def test_product_user_max_order_amount_no_restriction(self):
+        product = models.Product.objects.create(
+            name="Test no restriction", max_allowed_per_shift=None, current_price=0.8
+        )
+        self.assertEqual(product.user_max_order_amount(self.normal_user, self.shift), None)
+        models.Order.objects.create(user=self.normal_user, shift=self.shift, product=product)
+        self.assertEqual(product.user_max_order_amount(self.normal_user, self.shift), None)
+        models.Order.objects.create(user=self.normal_user, shift=self.shift, product=self.product_2)
+        self.assertEqual(product.user_max_order_amount(self.normal_user, self.shift), None)
+
+    def test_active_venue_validator(self):
+        self.assertTrue(self.order_venue.venue.active)
+        self.assertTrue(models.active_venue_validator(self.order_venue))
+        self.order_venue.venue.active = False
+        self.order_venue.venue.save()
+
+        def do_raise_validation_error():
+            models.active_venue_validator(self.order_venue)
+
+        self.assertRaises(ValidationError, do_raise_validation_error)
```

Comment:
Same here, just pass the arguments as extra params instead of declaring an inline function